install/gcp/installer/util/utils.py

```diff
@@ -232,37 +232,57 @@ def infer_version(advanced_mode):
     Returns:
         str: Selected Forseti branch.
     """
+    version = None
     return_code, out, err = run_command(
-        ['git', 'rev-parse', '--abbrev-ref', 'HEAD'])
+        ['git', 'symbolic-ref', '-q', '--short', 'HEAD'],
+        number_of_retry=0)
+    # The git command above will return empty if the user is not currently
+    # on a branch.
     if return_code:
         print(err)
         print('Will try to infer the Forseti version instead.')
     else:
-        branch = out.strip()
+        version = out.strip()
+
+    # For the installation to work, user must be on a tag or a branch.
+    # If the user is not on a branch, we check whether or not the user
+    # is on a tag, if not, we will exit the installation.
+    if not version:
+        return_code, out, err = run_command(
+            ['git', 'describe', '--tags', '--exact-match'])
+        # The git command above will return the tag name if we checked out
+        # a tag, will throw an exception otherwise.
+        if return_code:
+            print(err)
+            print('Unable to determine the current Forseti version, please '
+                  'check https://forsetisecurity.org/faq/ for more '
+                  'information.')
+            sys.exit(1)
+            version = 'tags/{}'.format(out.strip()) if out.strip() else ''
```

Comment:
Is this right, to be inside the `if return_code` block?

Reply:
Updated.